R/helper_lm_robust_fit.R

```diff
@@ -260,7 +260,11 @@ lm_robust_fit <- function(y,
       return_list[["fixed_effects"]] <- setNames(
         tapply(
           return_list[["fitted.values"]] -
-            data[["Xoriginal"]][, variable_names, drop = FALSE] %*% fit$beta_hat,
+            data[["Xoriginal"]][
+              ,
+              colnames(data[["X"]]),
```

Comment:
all_variable_names ? The intent would be clearer here if you updated all_variable_names at the same place where Xoriginal loses those columns. 

Let's also think about splitting up this 300 line function into something more manageable someday maybe. This is a good candidate for a helper.